docs/fpc-registration-old.puml

```diff
@@ -0,0 +1,83 @@
+@startuml
+
+'- force wrapping of longer text
+'skinparam wrapWidth 400
+
+!pragma teoz true
+
+
+title Fabric Private Chaincode - Enclave registry - Current approach
+
+actor	Admin order 10
+participant ECC order 20
+participant ERCC order 40
+entity IAS order 50
+collections Orderer order 70
+collections Peers order 80
+
+note right of Peers
+  **Drawbacks**
+  - Custom validation at __every__ peer
+  - Single-endorser policy only
+  - ERCC must contact IAS (internect connection and IAS credentials needed)
```

Comment:
Crucially also we currently have no way to securely pass the credentials to ERCC only (i.e., all chaincode running on that peer does receive the credentials ...)